modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,506 @@
+        String requestUrl = webAppURL + "/" + IS_WEB_APP_LOADED_METHOD_URL + "/" + tenantDomain + "/" + webAppName;
+        WebAppStatus webAppStatus;
+        try {
+
+            HttpResponse response = HttpURLConnectionClient.sendGetRequest(requestUrl, null);
+            JSONObject webAppStatusJSON = new JSONObject(response.getData()).getJSONObject("WebAppStatus");
+            boolean isTenantLoaded = webAppStatusJSON.getJSONObject("tenantStatus").getBoolean("tenantContextLoaded");
+            boolean isWebAppStarted = webAppStatusJSON.getBoolean("webAppStarted");
+            boolean isWebAppGhost = webAppStatusJSON.getBoolean("webAppGhost");
+            webAppStatus = new WebAppStatus(new TenantStatus(isTenantLoaded), isWebAppStarted, isWebAppGhost);
+        } catch (IOException ioException) {
+            String customErrorMessage = "IOException when sending the Get request to:" + requestUrl + "\n" +
+                    ioException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, ioException);
+        } catch (JSONException jsonException) {
+            String customErrorMessage = "JSONException when retrieving the values from json object WebAppStatus" +
+                    requestUrl + "\n" + jsonException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, jsonException);
+        }
+
+        return webAppStatus;
+
+    }
+
+
+    /**
+     * Check the given Jaggery application is deployed correctly. This method is wait for 90 seconds
+     * for deployment of jaggery application and each 500 milliseconds  of wait it will check the
+     * deployment status.
+     *
+     * @param appName Name of the application.
+     * @return true if the application is get deployed before the maximum wait time of 90 seconds.
+     * @throws LazyLoadingTestException
+     */
+    protected boolean isJaggeryAppDeployed(String appName) throws LazyLoadingTestException {
+        int deploymentDelay = 90 * 1000;
+        log.info("waiting " + deploymentDelay + " millis for Service deployment " + appName);
+        WebAppAdminClient webAppAdminClient;
+        try {
+            webAppAdminClient = new WebAppAdminClient(backendURL, sessionCookie);
+        } catch (AxisFault axisFault) {
+            String customErrorMessage = "AxisFault Exception  when creating WebAppAdminClient object. backend URL:" +
+                    backendURL + " Session Cookie: " + sessionCookie + "\n" + axisFault.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, axisFault);
+        }
+        List<String> webAppList;
+        List<String> faultyWebAppList;
+
+        boolean isWebAppDeployed = false;
+        Calendar startTime = Calendar.getInstance();
```

Comment:
shall we stick to the system time rather depending on calendar class 

System.currentTimeMillis() - 

see - http://stackoverflow.com/questions/368094/system-currenttimemillis-vs-new-date-vs-calendar-getinstance-gettime